File: rust/2000.rs
impl Solution {
    pub fn reverse_prefix(word: String, ch: char) -> String {
        match word.find(ch) {
            Some(i) => {
                format!("{}{}", &word[..i + 1].chars().rev().collect::<String>(), &word[i + 1..])
            }
            None => {
                word
            }
        }
    }
}

File: go/2000.go
func reversePrefix(word string, ch byte) string {
    j := 0
    for idx, c := range word {
        if byte(c) == ch {
            j = idx
            break
        }
    }
    if j != 0 {
        r := []rune(word)
        for i := 0; i < j; i, j = i + 1, j - 1 {
            r[i], r[j] = r[j], r[i]
        }
        return string(r)
    }
    return word
}
